.github/workflows/npm-publish.yml

```diff
@@ -0,0 +1,30 @@
+name: Node.js Package
+
+on:
+  release:
+    types: [created]
```

Comment:
So let's say I create a release page on GitHub named "v1.0.0-alpha.1" and I also create a tag named "v1.0.0-alpha.1" but the version in my `package.json` is "1.0.0". GitHub will release version "1.0.0" or "1.0.0-alpha.1"?
In other words, do I have to perform a "npm version" to update the version in the package.json first? If so, I think it's better to trigger a release on a tag:

```
on:
  push:
    tags: 
      - 'v*'
```

So the workflow would be:

```
$ npm version v1.0.0.alpha.1
$ git push origin master --tags
```

(edit the release page on GitHub).

I've never used GitHub actions so I might be wrong :sweat_smile:

Reply:
`npm publish` publishes a version as stated in the `package.json` file.

I understand `npm version xxx` as a way to sync the Git tag with the `package.json#version` field.

Creating a release also creates a tag — it's the same thing except the "release" stuff are non-Git related, but GitHub related.

So maybe to be explicit, it would make more sense to use your suggested `on.push.tags` syntax.